src/pda/tests/pda.service.spec.ts

```diff
@@ -0,0 +1,235 @@
+import { Test, TestingModule } from "@nestjs/testing";
+import { PDAService } from "../pda.service";
+import { WinstonProvider } from '@common/winston/winston.provider';
+import { HttpModule, HttpService } from '@nestjs/axios'; 
+import { ConfigModule, ConfigService } from '@nestjs/config'; 
+import { AxiosResponse } from 'axios';
+import { IssuedPDA } from '../interfaces/pda.interface';
+import { of } from 'rxjs';
+
+
+
+jest.mock('@common/winston/winston.provider');
+
+
+describe('PDAService', () => {
+  let service: PDAService;
+  let axios: HttpService;
+  let config: ConfigService;
+
+
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [HttpModule, ConfigModule],
+      providers: [WinstonProvider, PDAService],
+    }).compile();
+
+    service = module.get<PDAService>(PDAService);
+    axios = module.get<HttpService>(HttpService);
+    config = module.get<ConfigService>(ConfigService);
+
+    jest.clearAllMocks();
+  });
+
+  test('Should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  describe('When the request method called', () => {
+    const query = 'query';
+    let axiosResponse: AxiosResponse;
+    let variables: Record<string, any>;
+    let returnValue: Record<string, any>;
+    
+
+    beforeEach(async () => {
+      axiosResponse = {
+        data: { issuedPDAs: [{ 'pda': 'pda' }] },
+        status: 200,
+        statusText: 'OK',
+        headers: undefined,
+        config: undefined,
+      };
+
+
+      jest.spyOn(config, 'get').mockReturnValue('');
+      jest.spyOn(axios, 'post').mockReturnValue(of(axiosResponse));
+      jest.spyOn(axios, 'request').mockReturnValue(of(axiosResponse));
+
+
+      returnValue = await service['request'](query, variables);
+
+    });
+    test('Should be defined', () => {
+      expect(service['request']).toBeDefined();
+    });
+
+
+    test('Should call post from axios with the correct parameters', () => {
+      expect(axios.post).toHaveBeenCalledWith(
+        '',
+        {
+          query: 'query',
+          variables: { key: 'value' },
+        },
+        {
+          headers: {
+            Authorization: 'Bearer ',
+            'x-api-key': '',
+            'Content-Type': 'application/json',
+          },
+        },
+      );
+    })
+
+    test('Should call get method from config', () => {
+      expect(config.get).toHaveBeenCalledWith('MYGATEWAY_ENDPOINT_URL');
+    });
+
+    test('Should call get method from config', () => {
+      expect(config.get).toHaveBeenCalledWith('MYGATEWAY_AUTHENTICATION_TOKEN');
+    });
+
+    test('Should call get method from config', () => {
+      expect(config.get).toHaveBeenCalledWith('MYGATEWAY_API_KEY');
+    });
+
+
+
+
+
+
+    test('Should return the data from the http response', () => {
+      expect(returnValue).toEqual(axiosResponse.data);
+    });
+
+
+  })
+  describe('When the getIssuedPDAsGQL method called', () => {
```

Comment:
you should not execute the logic several times! you have to call it first and write tests for it.